src/participant.ts

```diff
@@ -406,3 +435,100 @@ export class Participant extends EventEmitter {
     this.frame.removeEventListener('load', this.onFrameLoad);
   };
 }
+
+/**
+ * Represents raw data received from the interactive server.
+ */
+interface IIncomingPacket {
+  type: string;
+  method: string;
+  params: any;
+}
+
+interface IScene {
+  sceneID: string;
+  controls: IControl[] | null;
+}
+
+interface IControl {
+  controlID: string;
+  kind: string;
+}
+
+interface IGroup {
+  groupID: string;
+  sceneID: string;
+}
+
+/**
+ * Retains data required to acquire a control's kind.
+ */
+class ControlsState extends EventEmitter {
```

Comment:
We have the `dumpState` method on the participant to avoid needing to do this

Reply:
Ooo didn't realise that was a thing. :) Looks like this would require integration developers to implement it -- the preact starter does this, but I'm not sure how we'd handle this for developers that don't implement it?